### ISSUE TITLE
Why is this not in the logstash-plugins organisation?

### ISSUE THREAD

**Issue by theangryangel** [theangryangel/logstash-output-jdbc#31]
I've been asked this several times over various media, and I'm getting a bit fed up of answering this.

I'll be frank, I'm not overly fond of the logstash-plugins.
- Many plugins there seem to have a standard template README which are useless or wrong:
  - For instance describing the plugin as being a filter when it's an output plugin - a minor point I'll concede
  - Nothing else about what it does or what it's capabilities, drawbacks or limitations are
  - No example configurations
- I find the auto-generated documentation on elastic.co to be border line acceptable in most of the plugins, which is something I find a bit perturbing since Jordan effectively described documentation as a first class bug in his early talks on logstash. ~~I'll admit that a lack of tests in this repo is probably borderline as bad as little/no documentation, however I do use this plugin in production myself and I have a vested interest in not breaking it. As and when I have time I will sort the lack of tests~~
- I don't trust elastic.co: I find their frequent naming, version scheme, and repository alterations irritating. Lumberjack is a prime example. Then it had a name change. Then it moved to one domain. Then it moved to another domain. Then it got switched out for Beats. I'm all for improving a product, but if you don't live in the ELK eco-system this can and will bite you in the ass

I've marked this as wontfix for now. I'd be happy to see my opinion changed.

**Comment by jordansissel** [comment]
I'm happy to discuss your concerns, if you like --

You points are all well made, and I agree. If you want background on why any given decisions (plugin README structure, lumberjack rename, etc) were made, I am happy to provide, but for brevity I'll focus on what we can do going forward -

Maintainers of plugins in the logstash-plugins repos are full owners of their repositories. Like your own git repos, maintainers are welcome to write whatever documentation they feel is appropriate, in whatever places -- including README or plugin documentation that would appear on elastic.co.

Our encouragement to have plugins centrally in the logstash-plugins org is aimed at benefiting users and developers. I agree, again, with your concerns, and we are working towards improving things.

As for documentation, I'm not sure what specifically you find poor about the plugin docs today, but I'm open to hearing about it. Most of the content for each plugin is written by humans, not machines -- though it is organized by machines.

The thing we try to provide with the logstash-plugins github org is a common gateway for us to discover what plugins should be aware of when it comes to integration tests and publishing of documentation. We cannot, and will not, try to force you in any way to move this repo to logstash-plugins, and there are no wrong choices to make, here. :)

**Comment by theangryangel** [comment]
Thanks for dropping by @jordansissel.

I will agree that in 2.x the documentation is better than it was. It's been sometime since I've really taken any notice of it, and my thoughts are outdated. My environment more or less Just Works(TM), and my needs haven't changed, so I've not really looked at it.

There are a few things I think could be improved, largely around the community plugins.
- Plugins with no documentation like metaevent - I have no idea what that does. Might be useful, but honestly I'm too lazy to look at the code. I get that logstash-plugins is largely community maintained, but stuff like that makes me wonder about the quality of community code, and if logstash-plugins as an org tries to strive for the same goals you set out to.
I'm not saying what I've produced in this repo is great (it's really not - it works well enough for my needs, and hopefully a few other people).
- Example configurations, especially for plugins which can interact with a wide number of other systems (like this one) have reduced the amount people have reaching out to me
-  Every filter has the standard filter features (add/remove_tag, etc.) listed. Having them separated or linked to a standard features page, or something would make the page easier to scan exactly what I'm looking for, in my opinion.

I've been thinking over Alvin's email over the weekend, but I'm really undecided now. Ultimately my issues over documentation are really with other community plugins, and there's not much anyone can do about that. 

I'd like to see this plugin get a bit more visibility (I actually met a guy who was implementing it independently a few months back because it wasn't in the org), and logstash-plugins is a way to prevent any more duplicate work. I totally see the benefit. 

I guess my only other worry is keeping up with elastic.co and where-ever logstash might be heading.

I'll have more of a think tonight and come back to you guys.

**Comment by antgel** [comment]
Ping to see if anyone has any thoughts on this...

**Comment by theangryangel** [comment]
My thoughts on elastic as a company hasn’t really changed much.

A few months back I asked publicly how to get more information on what’s coming through the pipeline, etc without having to be constantly immersed in the logstash ecosystem. I got complete silence. 

The last time I looked at the logstash-plugins org I saw a fair few dead plugins asking for maintainers. The only benefit would be visibility, and maybe if I’m lucky elastic would assist in some small parts. But given how non core plugins seem to die a death there, I’d guess not? 

If I’m going to continue to be a part time maintainer extra visibility right now isn’t something I really want. I’m burnt out on support requests for the most basic of things like user names and passwords being incorrect, feature requests for really company specific use cases, and even people not understanding what logatash is.
